client/tests/test_serving_model.py

```diff
@@ -0,0 +1,195 @@
+import featureform as ff
+import os
+import pytest
+
+
+real_path = os.path.realpath(__file__)
+dir_path = os.path.dirname(real_path)
+
+# TRAINING SET TESTS
+@pytest.mark.parametrize(
+    "provider_source_fxt,resource_serving_client_fxt,is_local",
+    [
+        ('local_provider_and_source', 'resource_serving_clients', True),
+        ('hosted_sql_provider_and_source','resource_serving_clients', False),
+    ]
+)
+def test_registering_model_while_serving_training_set(provider_source_fxt,resource_serving_client_fxt, is_local, request):
+    provider, source = request.getfixturevalue(provider_source_fxt);
+    resource_client, serving_client = request.getfixturevalue(resource_serving_client_fxt)(is_local);
+
+    # Arranges the resources context following the Quickstart pattern
+    arrange_resources(provider, source, resource_client, is_local)
+
+    model_name = 'fraud_model_a';
+
+    serving_client.training_set("fraud_training", "quickstart", model=model_name)
+
+    model = resource_client.get_model(model_name)
+
+    assert model.name() == model_name
+
+
+@pytest.mark.parametrize(
+    "provider_source_fxt,resource_serving_client_fxt,is_local",
+    [
+        ('local_provider_and_source', 'resource_serving_clients', True),
+        ('hosted_sql_provider_and_source','resource_serving_clients', False),
+    ]
+)
+def test_registering_two_model_while_serving_training_set(provider_source_fxt,resource_serving_client_fxt, is_local, request):
+    provider, source = request.getfixturevalue(provider_source_fxt);
+    resource_client, serving_client = request.getfixturevalue(resource_serving_client_fxt)(is_local);
+
+    # Arranges the resources context following the Quickstart pattern
+    arrange_resources(provider, source, resource_client, is_local)
+
+    model_name_a = 'fraud_model_a';
+    model_name_b = 'fraud_model_b';
+
+    serving_client.training_set("fraud_training", "quickstart", model=model_name_a)
+    serving_client.training_set("fraud_training", "quickstart", model=model_name_b)
+
+    models = resource_client.list_models(is_local)
```

Comment:
Should have a test case for empty `list_models` unless I missed it somewhere else